rs/replicated_state/src/replicated_state.rs

```diff
@@ -957,6 +969,94 @@ impl ReplicatedState {
         (expired_callback_count, errors)
     }
 
+    /// Enforces the best-effort message limit by repeatedly shedding the largest
+    /// best-effort messages of the canisters with the largest best-effort memory
+    /// usage until the memory usage drops below the limit.
```

Comment:
This seems imprecise. I had to really look at the code to understand what exactly this means.

IIUC the priority queue holds the canisters by memory usage, i.e. the one with the largest usage in front. It then sheds the largest message from this canister until we either go below the limit or a different canister becomes the one with the largest memory usage (or we run out of messages altogether). So it's not some sort of round robin scheme over the _**canisters with the largest best-effort memory usage**_ where each has its largest message shed or something (which I was sort of guessing from reading the comment).

Reply:
Changed it to say "repeatedly shedding the largest best-effort message of the canister with the highest best-effort message memory usage" (i.e. dropped the plurals).

If you have a better suggestion, I'm happy to apply it.